GTest

roigcarlo opened the pull request:
**📝 Description**
Implementation of GTests.

### Depending on:

- [x] #12317
- [x] #12306

### Tasks

- [x] Fix exit code -11 appearing on some suites
- [x] Fix windows not executing any test 
- [x] Fix `ubuntu-intel` reporting exit code as valid, and tests reporting false positives.

### Related

- [x] #12326 (Currently Skipped)

### Kernel
Kernel changes are being discussed in separate PR's (#12281, #12306). This PR enabled the de-registration of components.

### Testing Mechanism
All cpp can now be run directly from `bin/Release/tests/[SuiteName]`. In order to tests and debug there are several new options like:
- `--gtest_filter="foo"`: To filter tests by name. Accepts wildcards.
- `--gtest_break_on_failure`: To tell to stop in the first failure
- `--gtest_catch_exceptions=0`: To prevent from capturing exepctions and be able to hook gdb directly on the faulty instruction

Complete list can be found in the gtest documentation.

Tests als can be run through the testing scripts as usual:
- `testing/run_python_tests.py`
- `testing/run_python_mpi_tests.py`
- `testing/run_cpp_tests.py`
- `testing/run_cpp_mpi_tests.py`

### Compilation Targets 
Tests are now compiled in the majority of the cases with the `kratos_add_gtests` macro. As example:

```CMAKE
## CableNet Testing sources
if(${KRATOS_BUILD_TESTING} MATCHES ON)
    file(GLOB_RECURSE KRATOS_CABLE_NET_APPLICATION_TESTING_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/tests/*.cpp)
    kratos_add_gtests(TARGET KratosCableNetCore SOURCES "${KRATOS_CABLE_NET_APPLICATION_TESTING_SOURCES}" USE_CUSTOM_MAIN)
endif(${KRATOS_BUILD_TESTING} MATCHES ON)
```

This macro accepts several options:
- `TARGET`: The target name of the library you will tests. In 99% this will be `Kratos[your_app_name]Core`. It will create a  `Kratos[your_app_name]CoreTest` as result.
- `SOURCES`: The list of the cpp files containing your tests and possible utilities needed by them. From now on we will not accept testing utilities as part of the core libraries in order to reduce the weight of the binaries.
- `USE_MPI`: Added this flag if you target depends on MPI (MPICore, Trilinos, MPIExtensions, etc...)
- `USE_CUSTOM_MAIN`: Add this flag if you have defined a custom main for your tests. In this PR this flag is enabled for most of the applications as a transition mechanism to keep the PR as small as possible. More in the transition section.

There are new several compilation targets as a result of the changes in the tests:

```
(lib) KratosCore            <- Main Kratos functionality
(lib) KratosApplication     <- Python Interface (Depends on KratosCore)
(lib) KratosTestingUtils    <- Auxiliar functions for testing (Depends on KratosCore)
(exe) KratosTests           <- The actual tests (Depends on  KratosTestingUtils, KratosCore)
(lib) KratosMPICore         <- Main MPI Kratos functionality
(lib) KratosMPITestingUtils <- Auxiliar functions for testing (Depends on KratosMPICore, KratosCore)
(exe) KratosMPITests        <- The actual tests (Depends on  KratosMPITestingUtils, KratosMPICore, KratosTestingUtils, KratosCore)
```

In general you should not bother about those unless you want to do special things with your applications.

### Transition Changes
Due to some changes in the structure of how the targets are compiler, tests no longer depend on the `testing` sources from Kratos. Specifically a change that will be required is to change "include/testing.h" to an include of you application testing suit.

As this would imply changing all testing sources from Kratos in a single PR, most of applications have a transition solution in which a generic suit is defined in the kernel and the includes are the same. A placeholder is provided as well which will need to be changed to the final solution.

StructuralMechanics, Geomechanics, CoSimulationApplication and MappingApplication have the new mechanism in which the suite is included directly from the application. We expect to replace the rest of applications in future PR once this is merged.


**🆕 Changelog**
- Adding GTests to Kratos

roigcarlo committed:
Prepare to build gtests

roigcarlo committed:
Removing references to runkratos

roigcarlo committed:
Enable gtesting

roigcarlo committed:
Add target for the core

roigcarlo committed:
Moving mpi related test utilities out of the core

roigcarlo committed:
Fixing mpi dependant tests

roigcarlo committed:
Refactor core and mpicore tests

roigcarlo committed:
Updating tests

roigcarlo committed:
Adding gtest

roigcarlo committed:
Patch for cmake 3.27

roigcarlo committed:
Mpi tests

roigcarlo committed:
Adding kernel destructor and fixing mpi test with individual kernel instances

roigcarlo committed:
Restoring internal macros checks

roigcarlo committed:
Fixing problem comparing array_1d with ZeroVector(x) from boost

roigcarlo committed:
Fixing cstr cmp in text excpetion

roigcarlo committed:
Fixing cstr cmp in code location

roigcarlo committed:
Test_checks now checks the checks class. Checkmate

roigcarlo committed:
Fixing logger test (bad copy paste)

roigcarlo committed:
Merge branch 'master' into ci/gtests

roigcarlo committed:
Min cmake version

roigcarlo committed:
Missing include

roigcarlo committed:
Using distributed expects

roigcarlo committed:
Merge branch 'master' into ci/gtests

loumalouomega commented on kratos/tests/test_utilities/test_laplacian_element.cpp:
```diff
@@ -20,7 +20,7 @@
 #include "utilities/math_utils.h"
 #include "utilities/geometry_utilities.h"
-#include "tests/cpp_tests/auxiliar_files_for_cpp_unnitest/test_laplacian_element.h"
+#include "tests/test_utilities/test_laplacian_element.h"
```
Can you move these files in an independent PR?

roigcarlo replied:
I would prefer not to, but if its strictly necessary I can.

loumalouomega replied:
This PR is humngous, it will simplify a lot review

roigcarlo replied:
Ok I'll try, but this is only like the base of the PR. I will have to change all the apps as well. I will prepare smaller PR to merge to this one but there is no way that both systems coexists :S so changes need to be in a single PR

loumalouomega commented on scripts/wheels/linux/configure.sh:
```diff
@@ -75,5 +75,4 @@ cmake -H"${KRATOS_SOURCE}" -B"${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}" \
 -DINCLUDE_MMG=ON                                                       \
 -DMMG_ROOT="/workspace/external_libraries/mmg/mmg_5_5_1"               \
 -DKRATOS_BUILD_TESTING=OFF                                             \
--DINSTALL_RUNKRATOS=OFF                                                \
```
Everything related with RunKratos should go in a independent PR?

loumalouomega commented on kratos/tests/cpp_tests/utilities/test_variable_utils.cpp:
```diff
@@ -59,12 +59,13 @@ KRATOS_TEST_CASE_IN_SUITE(VariableUtilsSetHistoricalVariablesToZero, KratosCoreF
-        KRATOS_CHECK_VECTOR_NEAR(r_node.FastGetSolutionStepValue(VELOCITY), ZeroVector(3), tolerance);
-        KRATOS_CHECK_VECTOR_NEAR(r_node.FastGetSolutionStepValue(DISPLACEMENT), ZeroVector(3), tolerance);
-        KRATOS_CHECK_MATRIX_NEAR(r_node.FastGetSolutionStepValue(DEFORMATION_GRADIENT), ZeroMatrix(0,0), tolerance);
+        KRATOS_EXPECT_NEAR(r_node.FastGetSolutionStepValue(PRESSURE), 0.0, tolerance);
```
It was really necessary to change the checks?, it would be smarter to keep them and calling the new ones, and remove bit a bit, otherwise this comprieses many changes

roigcarlo replied:
Yep, this check specifically fails if trying to compare boost vectors (see https://github.com/KratosMultiphysics/Kratos/pull/11506/commits/8c34e8893df63d25c803ecbb542b75758b4a317b)

loumalouomega replied:
Okay

loumalouomega replied:
Can we keep a transition interface?, think thatthis will break many branches and independent projects using Kratos

roigcarlo replied:
I don't see how, I've tried. As I commented in the @KratosMultiphysics/technical-committee there is currently missuse of the `KRATOS_CHECK_XXX` macros, as they are used both as a pseudo-assertions in the tests, and as checks in the rest of the code. 

Since gtest uses full fledged assertions for the test checks, I cannot use the same macro for both, so either we change the KRATOS_CHECK in the code to something else (either be another macro or a proper check) or we make another interface for the assertions based on the existing one (what I did)

If you see any other solution tell me and I will implement it.

loumalouomega replied:
In that case we can change the macro from CHECK to ASSERT in a independent PR, so this PR is not 250 changes and that way other projects can be adapted

roigcarlo replied:
The problem is that it will still be a 250+ files changed PR, but as you like.

loumalouomega replied:
Yes, but it will be a simple change-> Just renaming macro. Here we have renaming + gtest + runkratos + ...

loumalouomega commented on kratos/testing_/test_skipped_exception.h:
```diff
@@ -0,0 +1,129 @@
+//    |  /           |
```
`testing_` folder?

loumalouomega replied:
I see many renamings because of this

roigcarlo replied:
This is leftover, I will remove it, don't worry. Its there because I still need to use things from the older testing.

loumalouomega commented on kratos/sources/kernel.cpp:
```diff
@@ -36,6 +36,36 @@ Kernel::Kernel(bool IsDistributedRun) : mpKratosCoreApplication(Kratos::make_sha
 }
 
+Kernel::~Kernel() {
+    KratosComponents<Variable<bool>>::Clear();
```
Why this?, independent PR?

roigcarlo replied:
This is needed so the fixtures work correctly. I will detail it in the PR description

loumalouomega replied:
Can be done also in a indpendent PR

loumalouomega replied:
Please otherwise will be impossible to check properly

roigcarlo replied:
Agree and not. If I make a separate PR for this you will want a test, and the test can only be done if we do this. It's a snake chasing his tail :S

loumalouomega commented on kratos/python/add_testing_to_python.cpp:
```diff
@@ -15,56 +15,56 @@
+	// namespace py = pybind11;
 
-    py::class_<Testing::Tester, Kratos::shared_ptr<Testing::Tester> > TesterPyBind(m, "Tester");
+    // py::class_<Testing::Tester, Kratos::shared_ptr<Testing::Tester> > TesterPyBind(m, "Tester");
```
Remove if unused

loumalouomega commented on kratos/includes/expect.h:
```diff
@@ -0,0 +1,211 @@
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
```
Tabs!

roigcarlo committed:
Removing testing_ folder

roigcarlo committed:
Merge branch 'master' into ci/gtests

roigcarlo committed:
Merge conflicts

roigcarlo committed:
Merge conflicts 2

roigcarlo committed:
Fix macros

roigcarlo committed:
Fix macros

roigcarlo committed:
Print info in function

roigcarlo committed:
Fixing mpi tests 1

roigcarlo committed:
Adding tests to fluid

roigcarlo committed:
Temp adding fixture to core

roigcarlo committed:
Merge remote-tracking branch 'origin' into ci/gtests

roigcarlo committed:
Decoupling the utilities from the kernel AGAIN

roigcarlo committed:
More includes and temporaly disabling exposure of testing utilities to the python layer through the core

roigcarlo committed:
Fix compilation problems and moving files back and forth

roigcarlo committed:
Add utility to autogenerate the test binaries for applications

roigcarlo committed:
Change CMakeLists of all applications with cpp test

Friendly and hidden reminder that applications not in this commit should be.
Unless I have accidentaly missed the changes :)

roigcarlo committed:
Adding mpi_expect.h

roigcarlo committed:
Missing change for FluidDynamics

roigcarlo committed:
Clarify why this is needed

roigcarlo committed:
Initial windows configuration

roigcarlo committed:
Typo

roigcarlo committed:
Fixing invoking macro

roigcarlo committed:
Setting export flag for testing tatgets

roigcarlo committed:
Missing test factories

roigcarlo committed:
Missing api expose for some geoclasses

roigcarlo committed:
Missing fixture for Structural. More to come

roigcarlo committed:
Correcting utilities exposure and api placement

roigcarlo committed:
Missing exports

roigcarlo committed:
Missing linking

roigcarlo committed:
Missing export api

roigcarlo committed:
Export api position consistency

roigcarlo committed:
Specifying visibility

avdg81 committed:
Moved geomechanics test suites to separate files

roigcarlo committed:
Adding possible fix for duplicated mpi messages on gtest

roigcarlo committed:
Using the correct fixture

roigcarlo committed:
Added missing redefines for pass and fail

roigcarlo committed:
Cosmetics

roigcarlo committed:
Cosmetics

Which are not cosmetics, because the gtest macro is no firend of
non-braced if/else

roigcarlo committed:
More non-bracer if/else corrected

rfaasse committed:
Minor fix in the `KRATOS_EXPECT_VECTOR_RELATIVE_NEAR` macro

roigcarlo committed:
fixing new include dirs

roigcarlo committed:
Making TrilinosCore dependencies public to propagate to CoreTest

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

roigcarlo committed:
Fixing mpi gtesting :) (TrilinosFixture pending)

roigcarlo committed:
Updated script for all applications

roigcarlo committed:
Fixing errors totaly not caused by me using a replace macro

roigcarlo committed:
Disabling previously added instantiation

rfaasse committed:
Fix missing import for geo tests and changed run_cpp_unit_tests.py for geo

rfaasse committed:
Fix missing import for structural tests and changed run_cpp_unit_tests.py for structural

roigcarlo committed:
Trying with dynamic build

roigcarlo committed:
Fixing structural quotes and splitting mapping mpi extension

rfaasse committed:
Merge remote-tracking branch 'origin/ci/gtests' into ci/gtests

roigcarlo committed:
Missing fixtures

roigcarlo committed:
Merge branch 'ci/gtests' of https://github.com/KratosMultiphysics/Kratos into ci/gtests

avdg81 committed:
Reimplemented the Python script `run_cpp_unit_tests.py`

- Removed the Kratos `import`s as they are no longer needed.
- The script's behavior is still the same (i.e. it still supports the
  `--onlyFastSuite` option); the underlying implementation, however, now
  calls the GTest test executable.
- Made the entry point and exit point symmetric.

roigcarlo committed:
Moving dll flag to root cmake

roigcarlo committed:
Moving dll flag to root cmake

roigcarlo committed:
Fixing sources for mpi extension

roigcarlo committed:
Restoring the old Testing communicator

roigcarlo committed:
Individual suites for applications. Some may be missing

avdg81 committed:
Some minor fixes

- Fixed an incorrect path in a GeoMechanicsApplication unit test.
- Uncommented an `#include` that was actually required.

roigcarlo committed:
Missing fixture, wrong include and enabled so option by default

roigcarlo committed:
Renaming wrong file

roigcarlo committed:
Restoring parallel env

roigcarlo committed:
Modify kernel initialization in the mpi fixture to be consistent with the core one

roigcarlo committed:
Moving kernel initializer to a helper class

roigcarlo committed:
Making trilinos tests work

rfaasse committed:
Fix export issue in variable utils

rfaasse committed:
Removing redundant '"' in geo CMakeLists.txt

roigcarlo committed:
Implementing tests globaly

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

avdg81 committed:
Only add a GTest target if there are any unit test source files

roigcarlo committed:
Separating mapping utilities from extension and fixing mapping compilation

roigcarlo committed:
Chanching the way linking happens to avoid problems in dependency resolution

roigcarlo committed:
Chanching default linking visibility of many symbols

roigcarlo committed:
Missing suite

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

roigcarlo committed:
Fixing bracers

roigcarlo committed:
I suppose this was a copy paste error

roigcarlo committed:
Incorrect name of the suit

roigcarlo committed:
nAvoid problems when mpi libraries cannot be discovered automatically due to the staged compilation

rfaasse committed:
Improved application suites for geo and structural mechanics

roigcarlo committed:
Several fixes

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

roigcarlo committed:
Adapting script for the new binaries

roigcarlo committed:
Excluding files because bad instantiation

roigcarlo committed:
Merge branch 'master' into ci/gtests

roigcarlo committed:
Fixed comflicting name while compiling with merged upstream

roigcarlo committed:
Starting to revamp runners for cpp tests

roigcarlo committed:
Ups :), fixed Mapping split which was done incorrectly

roigcarlo committed:
System imports first, one per line, lib imports seconds, and from X import Y after regular imports in the same order pleas dont put system imports AFTER project imports

roigcarlo committed:
More irrelevant files

roigcarlo committed:
Fixing RANS cpp tests

roigcarlo committed:
Updating cpp runner for serial tests

avdg81 committed:
Fixed some 'target_link_libraries' commands

roigcarlo committed:
Unifying testing scripts

roigcarlo committed:
Initial support for mpi

avdg81 committed:
Fixed a few includes and added an export

rfaasse committed:
Fix windows build for fluid and potential flow applications by adding KRATOS_API exports

roigcarlo committed:
Adding lib to path until gtest is installed on libs

avdg81 committed:
Fixed a linker error by exporting a class

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

roigcarlo committed:
Fixing centos

<insert sarcastic and mean comment about centos here>

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

avdg81 committed:
Fixed linker errors by exporting two classes

Also removed two files that were replaced by new ones.

avdg81 committed:
Forgot to add two new files

roigcarlo committed:
Fixing cpp report message

roigcarlo committed:
Extending the testing time

roigcarlo committed:
Taking the oportunity to fix anoying warnings

roigcarlo committed:
Moving default CMAKE_INSTALL_LIBS to libs to match kratos default lib install dir

roigcarlo committed:
Revert "Adding lib to path until gtest is installed on libs"

This reverts commit 97eef257bb37e0da7932a19f97ebed454fdc4084.

roigcarlo committed:
Rearanged files to move testing main outide of utilities and having output from tests hidden by default

roigcarlo committed:
Updating sources for tests and test_utilities targets

roigcarlo committed:
Fixing logger test (change in output line due to the change in the tests)

roigcarlo committed:
Fixing problem with unrgistered partial communicators being de-allocated by the test destructor

roigcarlo committed:
Restore the buffers after test TearDown

roigcarlo committed:
Moving implt to source

roigcarlo committed:
Missing file

roigcarlo committed:
Removing duplicated KRATOS_API

roigcarlo committed:
No running cpp tests automatically from the python interface anymore. Also this caused all of this tests to be ran twice¿?

roigcarlo committed:
Removing old run_cpp_tests (now run over the binary) and removing references to Tester from python layer

roigcarlo committed:
More references to old run_cpp

roigcarlo committed:
Fixing size of the system matrix for trilinos tests not currenly running on the CI

roigcarlo committed:
Fixing rans linking in windows

roigcarlo committed:
More run_cpp tests..

roigcarlo committed:
Renamed run_tests to run_python_tests and updated ci

roigcarlo committed:
Remove cpp from run_python_tests and cleanup subprocess

roigcarlo committed:
Forgot to add this in the fixing win api commit :P

roigcarlo committed:
Does not need to be exported

roigcarlo committed:
Fixing Cosim. Adding flag to be able to compile with custom main. Tests now compile with the kernel main by default

roigcarlo committed:
Merge branch 'master' into ci/gtests

roigcarlo committed:
Do not put utilities for testing in the core :)

rfaasse committed:
Enable settlement test, since every test now has their own kernel

rfaasse committed:
Fix linking errors on windows for meshing, mapping and iga application tests

avdg81 committed:
Fixed error when linking ContactStructuralMechanicsApplication on Windows

roigcarlo committed:
Removing last run_cpp

roigcarlo committed:
Remove tester

roigcarlo committed:
unifiying behaviour of testing scripts p1

roigcarlo committed:
Merge branch 'master' into ci/gtests

roigcarlo committed:
--using-mpi flag is now passed in the subprocess

roigcarlo committed:
Passing mpi command without spaces. Please avoid shell=true at all costs

roigcarlo committed:
Format using utf-8

roigcarlo committed:
consolidating cpp test runners

roigcarlo committed:
consolidating cpp test runners 2

roigcarlo committed:
Further consolidation of the testing scripts

Remoing code to try to deduce correct paths for applications. Live is
hard, follow conventions.
Re-Added core and core mpi using custom paths

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

roigcarlo committed:
Missing info

roigcarlo committed:
Fixing wring cast in the test

roigcarlo committed:
Keep comparison consistent @philbucher check if its ok

roigcarlo committed:
Disabling anoying print of the headers until verbosity is correctly implemented

roigcarlo committed:
Fixing dependency from structural

roigcarlo committed:
Setting runtime to 5 min

roigcarlo committed:
Enabling all tests always and importing Meshing app for mmg tests

roigcarlo committed:
Marking dependent tests as skip

roigcarlo committed:
Passig host file correctly

roigcarlo committed:
Unifying return codes and ensuring no overrides are pefromed in case the subprocess ends with something different than 0

roigcarlo committed:
No exit inside try

roigcarlo committed:
Fixing mpi-flags parsing and hopefully output coherence

roigcarlo committed:
Merge branch 'master' into ci/gtests

roigcarlo committed:
Capturing all exceptions and executing print regadless of the exit status

roigcarlo committed:
Temporaly disabling header print for shallow?

roigcarlo committed:
Temporaly disabling buffer redirections

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

roigcarlo committed:
Removing pragma once in source

roigcarlo committed:
Using relative paths

roigcarlo committed:
Fixing problems in mpi mapping

roigcarlo committed:
Merge branch 'master' into ci/gtests

roigcarlo committed:
Reverting asserts to check only distance

roigcarlo committed:
Revert "Using relative paths"

This reverts commit 80d89aedcbb58d596cb92d7fb7b04fcac3fec0eb.

roigcarlo committed:
Setting common runtime dir

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

roigcarlo committed:
disabling mpi_run_cpp

roigcarlo committed:
Passing run dir from launch script

roigcarlo committed:
Fixing tests not launching in win

roigcarlo committed:
Fixing mpi tests not launching in win (irrelevant, just in case)

roigcarlo committed:
Fixing install dir in win on some systems

avdg81 committed:
Disabled a failing C++ unit test in GeoMechanicsApplication

This test is unstable since it relies on file comparison. We'd better
come up with more robust test invariants prior to enabling the test.

roigcarlo committed:
Fixing string literal comparison

RiccardoRossi commented on applications/CoSimulationApplication/tests/cpp_distributed_tests/co_simulation_distributed_suite.cpp:
```diff
@@ -0,0 +1,27 @@
+// KRATOS  / ___|___/ ___|(_)_ __ ___  _   _| | __ _| |_(_) ___  _ ___
```
i understand that  these modifications have been agreed with  @philbucher correct?

RiccardoRossi commented on applications/ConstitutiveLawsApplication/tests/cpp_tests/constitutive_laws_fast_suite.h:
```diff
@@ -0,0 +1,29 @@
+  KratosConstitutiveLawsFastSuite();
+
+private:
+  KratosConstitutiveLawsApplication::Pointer mpConstitutiveLawsApp;
```
why do we need this but not the StructuralMechanics?

RiccardoRossi commented on applications/ContactStructuralMechanicsApplication/tests/cpp_tests/contact_structural_mechanics_fast_suite.cpp:
```diff
@@ -0,0 +1,34 @@
+    Kratos::Testing::mApplicationInitializerList.push_back([](std::vector<Kratos::KratosApplication::Pointer> & rRegisteredApplications, Kratos::Kernel & rKernel) {
+      if (!rKernel.IsImported("ContactStructuralMechanicsApplication")) {
+        auto pApplication = std::make_shared<Kratos::KratosContactStructuralMechanicsApplication>();
+        rKernel.ImportApplication(pApplication);
```
i am surprised that the StructuralMEchanics is not required by these tests.

RiccardoRossi commented on applications/IgaApplication/tests/cpp_tests/test_iga_shell_3p.cpp:
```diff
@@ -50,6 +50,8 @@ namespace Testing
     // Tests the stiffness matrix of the Shell3pElement with a polynomial degree of p=3.
     KRATOS_TEST_CASE_IN_SUITE(IgaShell3pElementP3, KratosIgaFastSuite)
     {
+        KRATOS_SKIP_TEST_IF_NOT(KratosComponents<ConstitutiveLaw>::Has("LinearElasticPlaneStress2DLaw")) << "This test needs the StructuralMechanicsApplication" << std::endl;
```
as i understand this means that these tests are currently being skipped. @rickyaristio @NickNick9 please take into account that they need to be moved to python ASAP

roigcarlo replied:
this is being tracked in https://github.com/KratosMultiphysics/Kratos/issues/12326

RiccardoRossi commented on kratos/tests/cpp_tests/testing/test_testing.cpp.disabled:
this is intentional or is it a leftover?

roigcarlo replied:
Leftover. Will fix

RiccardoRossi reviewed (commented):
some comments. I cannot claim this is a detailed review, nevertheless this looks good to me

roigcarlo committed:
Merge branch 'master' into ci/gtests

roigcarlo committed:
Enabling deregistration

roigcarlo committed:
Adding missing test suite

roigcarlo committed:
Remove leftover file

roigcarlo committed:
Updated tests to gtests

roigcarlo committed:
Remvoing old temporal deregister from kernel

roigcarlo committed:
Prevent unsigned values to mess up with the ordering check

roigcarlo committed:
More unsigned problems

roigcarlo committed:
Kratos API

roigcarlo committed:
Kratos API

roigcarlo committed:
Kratos API

roigcarlo committed:
disabling misterious error test

loumalouomega committed:
Merge branch 'master' into ci/gtests

loumalouomega committed:
Update base cmake with KRATOS_BASE_FOLDER

loumalouomega committed:
Using `KRATOS_BASE_FOLDER`

roigcarlo committed:
Remove comments

roigcarlo committed:
Initial doc

jcotela committed:
Move test to the correct suite

jcotela committed:
Remove uneeded suite fixture

jcotela committed:
Add class to handle importing runtime dependencies

jcotela committed:
Add some CreateApplication methods

jcotela committed:
Adding demo suite

jcotela committed:
Prefer KRATOS_COMPILED_IN_WINDOWS macro

jcotela committed:
Add CreateApplication for LinearSolversApplication

jcotela committed:
Update test suite example

jcotela committed:
Merge pull request #12475 from KratosMultiphysics/ci/gtest-runtime-dependency

[CI] Add support for importing runtime dependencies in gtest test suites

roigcarlo committed:
Reorganizing test structure

roigcarlo committed:
Fixing mistakes and comp problems

roigcarlo committed:
Fixing mpi changes

roigcarlo committed:
More mpi problems

jcotela committed:
Avoid including windows.h from a header, since it adds macros with conflicting names

jcotela committed:
Add missing KRATOS_API

roigcarlo committed:
Merge pull request #12486 from KratosMultiphysics/ci/gtests-dynload

Refactor of some aux testing files

roigcarlo committed:
Merge branch 'master' into ci/gtests

roigcarlo committed:
Update new tests

roigcarlo committed:
Missing exort api

roigcarlo committed:
Merge branch 'master' into ci/gtests

sunethwarna committed:
minor

roigcarlo committed:
Temporaly disabling test

roigcarlo committed:
Merge branch 'ci/gtests' of github.com:KratosMultiphysics/Kratos into ci/gtests

roigcarlo committed:
Temporaly disabling test for vectors
